### PR TITLE
Allow multiple security info on BS

### PR DESCRIPTION
**Pull request by phtrivier**
- Separated SecurityStore and BootstrapSecurityStore interfaces
- BootstrapSecurityStore supports checking multiple identities
  for a given endpoint
- Added identity check on bootstrap
- Added accessors for BootstrapConfig internal maps

Signed-off-by: Pierre-Henri Trivier <phtrivier@sierrawireless.com>

### CODE REVIEW

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/BootstrapTest.java [hunk 1]
```diff
@@ -49,7 +49,8 @@ public void stop() {
 
     @Test
     public void bootstrap() {
-        helper.waitForRegistration(2);
+        // FIXME(pht) wait 2 sec
```
Comment (jvermillard) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/BootstrapTest.java]: :scream:

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/security/BootstrapAuthService.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Sierra Wireless - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.security;
+
+import org.eclipse.leshan.core.request.Identity;
+
+public interface BootstrapAuthService {
```
Comment (jvermillard) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/security/BootstrapAuthService.java]: Javadoc :)

**Comment by sbernard31** [comment]
I think a PR or at list a commit by item would have been better.

 1. Added accessors for BootstrapConfig internal maps
 1. Added identity check on bootstrap
 1. Separated SecurityStore and BootstrapSecurityStore interfaces
 1. BootstrapSecurityStore supports checking multiple identities for a given endpoint

**Comment by sbernard31** [comment]
Sounds good for me.

**Comment by jvermillard** [comment]
merged:
* e4ecebf - Introduced BS authentication service (il y a 5 minutes) <Pierre-Henri Trivier>
* 0cdf7e1 - Separated BootstrapSecurityStore / SecurityStore (il y a 5 minutes) <Pierre-Henri Trivier>
* 9f30dd5 - Separated SecurityCheck (il y a 5 minutes) <Pierre-Henri Trivier>

**Pull request closed without merging**